lib/measures/tbd/resources/psi.rb

```diff
@@ -329,10 +373,36 @@ def processTBDinputs(surfaces, edges, set, io_path = nil, schema_path = nil)
             edge[:surfaces].each do |ss|
               match = false unless e[:surfaces].has_key?(ss)
             end
+
             next unless match
             if edge.has_key?(:length)      # optional, narrows down search (~1")
-              match = false unless (e[:length] - edge[:length]).abs < 0.025
+              match = false unless (e[:length] - edge[:length]).abs < 0.0254
             end
+
+            if edge.has_key?(:v0x) ||
+               edge.has_key?(:v0y) ||
+               edge.has_key?(:v0z) ||
+               edge.has_key?(:v1x) ||
+               edge.has_key?(:v1y) ||
+               edge.has_key?(:v1z)
+
+              unless edge.has_key?(:v0x) &&
+                     edge.has_key?(:v0y) &&
+                     edge.has_key?(:v0z) &&
+                     edge.has_key?(:v1x) &&
+                     edge.has_key?(:v1y) &&
+                     edge.has_key?(:v1z)
+                raise "Edge vertices must come in pairs"          # all or none
+              end
+              e1 = {}
+              e2 = {}
+              e1[:v0] = Topolys::Point3D.new(edge[:v0x].to_f, edge[:v0y].to_f, edge[:v0z].to_f)
+              e1[:v1] = Topolys::Point3D.new(edge[:v1x].to_f, edge[:v1y].to_f, edge[:v1z].to_f)
+              e2[:v0] = e[:v0].point
+              e2[:v1] = e[:v1].point
```

Comment:
What is `e`?  I don't quite get what is going on here

Reply:
You are trying to create new Topolys edges?

Reply:
`e `is an _TBD_ edge already held in memory (and whose vertices are Topolys::Point3D objects), while `edge` is one on file. So yes, the code here is generating temporary Topolys::Point3D strictly for comparison purposes.